ui/sd_internal/task_manager.py

```diff
@@ -219,26 +222,24 @@ def thread_get_next_task():
                     queued_task.error = Exception('cuda:0 is not available with the current config. Remove GFPGANer filter to run task.')
                     task = queued_task
                     break
-                if queued_task.request.use_cpu:
+                if queued_task.render_device == 'cpu':
                     queued_task.error = Exception('Cpu cannot be used to run this task. Remove GFPGANer filter to run task.')
                     task = queued_task
                     break
                 if not runtime.is_first_cuda_device(runtime.thread_data.device):
                     continue  # Wait for cuda:0
-            if queued_task.request.use_cpu and runtime.thread_data.device != 'cpu':
-                if is_alive('cpu') > 0:
-                    continue  # CPU Tasks, Skip GPU device
+            if queued_task.render_device and runtime.thread_data.device != queued_task.render_device:
```

Comment:
In the new logic, it looks like if the program started with a valid GPU (i.e. no thread started for CPU), the user will never be able to switch to CPU in a subsequent request.

It wouldn't throw an error previously, but now it will because `is_alive('cpu')` will be 0 (no thread was started for at the start).

Apologies if I've misunderstood how this works.